tests-integration/json-schema/test/json-schema.test.js

```diff
@@ -0,0 +1,202 @@
+/*
+    This file contains the automated tests for the JsonSchema backend.
+    It runs the make command to generate the IR from the Morphir models
+    Then it runs the gen command to generate the Json Schema.
+    The ajv library https://www.npmjs.com/package/ajv  is used to validate the generate json schema
+    and instances of the subschemas.
+*/
+// Imports
+const ajv2020 = require("ajv/dist/2020")
+const fs = require('fs')
+const util = require('util')
+const cli = require('../../../cli/cli')
+
+// Variables
+const basePath = "tests-integration/json-schema/model/"
+const schemaBasePath = "tests-integration/generated/jsonSchema/"
+const cliOptions = { typesOnly: false }
+const writeFile = util.promisify(fs.writeFile)
+var jsonObject
+const options = {
+    target : 'JsonSchema',
+}
+
+
+describe('Test Suite for Basic Types and Decimal',  () => {
+
+    beforeAll(async () => {
+        //Clear generated schema directory
+        fs.rmSync(schemaBasePath, {recursive: true, force : true})
+
+        //Create and write IR to file
+        const IR =  await cli.make(basePath, cliOptions)
+		await writeFile(basePath + 'morphir-ir.json', JSON.stringify(IR))
+
+        // Generate the Json schema
+        await cli.gen(basePath + 'morphir-ir.json', schemaBasePath , options)
+        const schemaPath = schemaBasePath + "TestModel.json"
+
+        // Read json into an object
+        const jsonBuffer = fs.readFileSync(schemaPath, 'utf8')
+        jsonObject = JSON.parse(jsonBuffer)
+	})
+
+    test('1. Bool type test case', () => {
+        const boolSchema= {
+            "type": "object",
+            "properties": {
+                "bool": {
+                    "$ref": "https://morphir.finos.org/test_model.schema.json#/$defs/BasicTypes.Paid"
+                }
+            },
+        }
+        expect(validatorBasic(boolSchema, {bool:true})).toBe(true)
```

Comment:
This does the right thing now but you could further improve it by putting more of the repetitive logic in `validatorBasic`. Here are a few more things I see that you could move inside:

- The whole `boolSchema` is the same for all basic types except for that small `BasicTypes.Paid` part.
- Adding the wrapper around the sample value could also be moved in. So you would just pass in `true` instead of `{bool:true}` into `validatorBasic`.